lib/common/issues/src/issue.rs

```diff
@@ -4,30 +4,53 @@ use chrono::{DateTime, Utc};
 use schemars::JsonSchema;
 use serde::Serialize;
 
+use crate::dashboard::Code;
 use crate::solution::Solution;
 
-/// Type of the issue code
-pub type CodeType = String;
-
 pub trait Issue {
-    fn code(&self) -> CodeType;
+    /// Differentiates issues of the same type
+    fn distinctive(&self) -> &str;
+
+    /// The codename for all issues of this type
+    fn name() -> &'static str;
+
+    /// A human-readable description of the issue
     fn description(&self) -> String;
+
+    /// Actionable solution to the issue
     fn solution(&self) -> Solution;
+
+    /// Internal code for the issue
+    fn code(&self) -> Code
+    where
+        Self: std::marker::Sized + 'static,
+    {
+        Code::new::<Self>(self.distinctive())
```

Comment:
it feels a bit weird to me to have a child field embedding part of the parent object.

Reply:
Good point. I'll move this to the `Code` implementation